js/navbar/tabEditor.js

```diff
@@ -45,8 +47,18 @@ const tabEditor = {
     if (showSearchbar !== false) {
       if (editingValue) {
         searchbar.showResults(editingValue, null)
+        searchbar.el.classList.remove('searchbar-hidden')
       } else {
         searchbar.showResults('', null)
+        const newTabUrlMode = settings.get('newTabUrl')
+        const isNTP = (newTabUrlMode === 'backgroundImage' && currentURL === '') ||
+                      (newTabUrlMode === 'customUrl' && currentURL === settings.get('newTabCustomUrl'))
+        
+        if (isNTP) {
+          searchbar.el.classList.add('searchbar-hidden')
```

Comment:
I think we could just skip calling `searchbar.showResults` in this case, and then we no longer need the hidden class.